File: app/src/main/java/com/bas080/autosleepdroid/AwakeDialogActivity.java
package com.bas080.autosleepdroid;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;

public class AwakeDialogActivity extends Activity {
    private AlertDialog alertDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        android.view.ContextThemeWrapper dialogContext = new android.view.ContextThemeWrapper(this, R.style.AppTheme);

        AlertDialog.Builder builder = new AlertDialog.Builder(dialogContext);
        builder.setTitle(R.string.dialog_awake_title);
        builder.setMessage(R.string.dialog_awake_message);
        builder.setPositiveButton(R.string.action_awake, (dialog, which) -> {
            Intent serviceIntent = new Intent(AwakeDialogActivity.this, SleepTimerService.class);
            serviceIntent.setAction(SleepTimerService.ACTION_AWAKE);
            if (android.os.Build.VERSION.SDK_INT >= 26) {
                startForegroundService(serviceIntent);
            } else {
                startService(serviceIntent);
            }
            finish();
        });
        builder.setNegativeButton(R.string.dialog_cancel, (dialog, which) -> finish());
        builder.setOnCancelListener(dialog -> finish());

        alertDialog = builder.create();
        alertDialog.setOnDismissListener(dialog -> {
            if (!isFinishing()) {
                finish();
            }
        });
        alertDialog.show();
    }

    public AlertDialog getAlertDialog() {
        return alertDialog;
    }
}

File: app/src/main/java/com/bas080/autosleepdroid/AwakeDialogActivity.kt
package com.bas080.autosleepdroid

import android.app.Activity
import android.app.AlertDialog
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.ContextThemeWrapper

class AwakeDialogActivity : Activity() {
    var alertDialog: AlertDialog? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val dialogContext = ContextThemeWrapper(this, R.style.AppTheme)

        val builder = AlertDialog.Builder(dialogContext)
        builder.setTitle(R.string.dialog_awake_title)
        builder.setMessage(R.string.dialog_awake_message)
        builder.setPositiveButton(R.string.action_awake) { _, _ ->
            val serviceIntent = Intent(this@AwakeDialogActivity, MainService::class.java).apply {
                action = MainService.ACTION_AWAKE
            }
            if (Build.VERSION.SDK_INT >= 26) {
                startForegroundService(serviceIntent)
            } else {
                startService(serviceIntent)
            }
            finish()
        }
        builder.setNegativeButton(R.string.dialog_cancel) { _, _ -> finish() }
        builder.setOnCancelListener { finish() }

        val dialog = builder.create()
        dialog.setOnDismissListener {
            if (!isFinishing) {
                finish()
            }
        }
        alertDialog = dialog
        dialog.show()
    }
}
